kotlinsample/src/main/java/com/airbnb/epoxy/kotlinsample/helpers/ViewBindingEpoxyModelWithHolder.kt

```diff
@@ -0,0 +1,55 @@
+package com.airbnb.epoxy.kotlinsample.helpers
+
+import android.view.View
+import androidx.viewbinding.ViewBinding
+import com.airbnb.epoxy.EpoxyHolder
+import com.airbnb.epoxy.EpoxyModelWithHolder
+import java.lang.reflect.Method
+import java.lang.reflect.ParameterizedType
+import java.util.concurrent.ConcurrentHashMap
+
+abstract class ViewBindingEpoxyModelWithHolder<in T : ViewBinding> :
+    EpoxyModelWithHolder<ViewBindingHolder>() {
+
+    @Suppress("UNCHECKED_CAST")
+    override fun bind(holder: ViewBindingHolder) {
+        (holder.viewBinding as T).bind()
+    }
+
+    abstract fun T.bind()
+
+    override fun createNewHolder(): ViewBindingHolder {
+        return ViewBindingHolder(this::class.java)
+    }
+}
+
+// Static cache of a method pointer for each type of item used.
+private val sBindingMethodByClass = ConcurrentHashMap<Class<*>, Method>()
+
+@Suppress("UNCHECKED_CAST")
+@Synchronized
+private fun getBindMethodFrom(javaClass: Class<*>): Method =
+    sBindingMethodByClass.getOrPut(javaClass) {
+        val actualTypeOfThis = getSuperclassParameterizedType(javaClass)
+        val viewBindingClass = actualTypeOfThis.actualTypeArguments[0] as Class<ViewBinding>
+        viewBindingClass.getDeclaredMethod("bind", View::class.java)
+            ?: error("The binder class ${javaClass.canonicalName} should have a method bind(View)")
+    }
+
+private fun getSuperclassParameterizedType(klass: Class<*>): ParameterizedType {
+    val genericSuperclass = klass.genericSuperclass
+    return (genericSuperclass as? ParameterizedType)
+        ?: getSuperclassParameterizedType(genericSuperclass as Class<*>)
+}
+
+class ViewBindingHolder(private val epoxyModelClass: Class<*>) : EpoxyHolder() {
```

Comment:
Can this holder class be parameterized with `<in T : ViewBinding>` too? that seems more accurate, since it should have the same type, and that may allow us to get the type directly from the holder instead of passing in `epoxyModelClass`

Reply:
Not sure how to do that actually, because of the generated code:
```
 * Generated file. Do not modify! */
public class ItemViewBindingEpoxyHolder_ extends ItemViewBindingEpoxyHolder implements GeneratedModel<ViewBindingHolder<T>>, ItemViewBindingEpoxyHolderBuilder {
  private OnModelBoundListener<ItemViewBindingEpoxyHolder_, ViewBindingHolder<T>> onModelBoundListener_epoxyGeneratedModel;
```
In the generated file, the 'T' is not defined as generic of the `ItemViewBindingEpoxyHolder_`.
From my limited understanding, I suppose it's because the processor doesn't handle ViewHolder with generics. Am I wrong? We could change that but I feel it's not that easy :)

Reply:
i see what you mean, this is fine :)